### DISCUSSION TITLE
Strongswan connection with free5gc issue

### DISCUSSION THREAD

**Discussion by joseph1267** [strongswan/strongswan#658] [Q&A] [answered]
Hi,
I have been implementing some codes in strongswan so that it can use 5g-aka to authenticate itself with the free5gc. 
Topology: **[ 10.0.0.8 --- 192.168.86.38 ] ===IPSEC tunnel=== [ 192.168.86.39 --- 10.0.0.10]**
                  strongswan + UE(10.0.0.8 VTI) VM                         free5gc VM (10.0.0.10 VTI)
And I was able to get to the point where ipsec tunnel establishment is done. After the ipsec tunnel is established, free5gc sends a NAS message to the strongswan side UE, a registration accept message. Ideally, it should be going from free5gc n3iwf (192.168.86.39) to subnet (VTI 10.0.0.10) and to the strongswan ipsec tunnel (192.168.86.38) and then at the end, to the UE (VTI 10.0.0.8). But the packets were not going from 192.168.86.39 to 10.0.0.10 in the first place. First I tried to ping from the 10.0.0.8 to the free5gc subnet (10.0.0.10) using ping -I ipsec0 10.0.0.10, and I was able to monitor the packets being sent. 10.0.0.8 -> 192.168.86.38 (At this point it will be encapsulated with ESP) - > 192.168.86.39 -x> 10.0.0.10. The packets were not going all the way to the end point. So, I thought it was the free5gc problem since it is between the 192.168.86.39 and 10.0.0.10, but I realized that strongswan is creating a xfrm state with espinudp, and free5gc was creating with tcp. So, I changed the free5gc to use espinudp in xfrm by adding some commend lines. And I was able to see the packets going all the way to the end point (10.0.0.8) from the other end point (10.0.0.10). But this does not solve the issue what I have, because I want to use free5gc's default xfrm that uses tcp connection to send NAS message. I have limited amount of knowledge about the ipsec and the xfrm policy and states, and may have done some serious changes that strongswan should not have been done. Please guide me to achieve my goal to send the NAS message to the UE using the TCP connection with the ipsec tunnel of strongswan. 
![free5gc](https://user-images.githubusercontent.com/45336417/136618690-5410cc3c-bf9e-46df-9143-74450a1368d7.PNG)
![image](https://user-images.githubusercontent.com/45336417/136618755-000fa631-1dc6-497b-9744-3637d7b14a18.png)

Thank you for reading.

**Comment by Thermi** [comment, marked as answer]
Unless you patched strongSwan to support it, it is not capable of configuring any SAs with ESP in TCP encapsulation.
